test/bdd/pkg/verifiable/verifiable_steps.go

```diff
@@ -103,38 +100,30 @@ func (s *SDKSteps) addVCProof(vc *verifiable.Credential, issuer, proofType strin
 	pubKey := doc.PublicKey[0]
 
 	kms := s.bddContext.AgentCtx[issuer].Signer()
-
-	s.proofType = proofType
+	signer := newSigner(kms, base58.Encode(pubKey.Value))
 
 	switch proofType {
 	case jwsLinkedDataProof:
-		verificationMethod := doc.ID + pubKey.ID
-
-		err := addJWSLinkedDataProof(vc, newSigner(kms, base58.Encode(pubKey.Value)), verificationMethod)
+		err := vc.AddLinkedDataProof(&verifiable.LinkedDataProofContext{
+			SignatureType:           "Ed25519Signature2018",
+			Suite:                   ed25519signature2018.New(ed25519signature2018.WithSigner(signer)),
+			SignatureRepresentation: verifiable.SignatureJWS,
+			Created:                 vc.Issued,
+			VerificationMethod:      doc.ID + pubKey.ID,
+		})
 		if err != nil {
 			return nil, err
 		}
 
 		return vc.MarshalJSON()
 
 	case jwsProof:
-		// TODO to be improved at https://github.com/hyperledger/aries-framework-go/issues/339:
-		//  1) use KMS for signing
-		//  2) do not keep public key, should be resolved by sidetree
 		jwtClaims, err := vc.JWTClaims(false)
 		if err != nil {
 			return nil, err
 		}
 
-		pubKey, privKey, err := ed25519.GenerateKey(rand.Reader)
-		if err != nil {
-			return nil, err
-		}
-
-		// keep pubKey to be used later for verification by a Holder
-		s.pubKey = pubKey
-
-		jws, err := jwtClaims.MarshalJWS(verifiable.EdDSA, privKey, "key-1")
+		jws, err := jwtClaims.MarshalJWS(verifiable.EdDSA, signer, pubKey.ID)
```

Comment:
Here we just pass `signer` (not a private key). Note that the same `signer` is used for LDP and JWS.